app/src/main/java/org/akvo/flow/presentation/geoshape/ViewGeoShapeActivity.java

```diff
@@ -0,0 +1,297 @@
+
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+
+import com.mapbox.geojson.Feature;
+import com.mapbox.geojson.FeatureCollection;
+import com.mapbox.geojson.Geometry;
+import com.mapbox.geojson.LineString;
+import com.mapbox.geojson.MultiPoint;
+import com.mapbox.geojson.Point;
+import com.mapbox.geojson.Polygon;
+import com.mapbox.mapboxsdk.camera.CameraUpdate;
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.geometry.LatLngBounds;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.Style;
+import com.mapbox.mapboxsdk.style.layers.CircleLayer;
+import com.mapbox.mapboxsdk.style.layers.FillLayer;
+import com.mapbox.mapboxsdk.style.layers.LineLayer;
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource;
+
+import org.akvo.flow.R;
+import org.akvo.flow.activity.BackActivity;
+import org.akvo.flow.util.ConstantUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import androidx.annotation.NonNull;
+import timber.log.Timber;
+
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.circleColor;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.circleRadius;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.circleStrokeColor;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.circleStrokeWidth;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.fillColor;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.lineColor;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.lineWidth;
+
+public class ViewGeoShapeActivity extends BackActivity {
+
+    private static final String CIRCLE_SOURCE_ID = "circle-source-id";
+    private static final String FILL_SOURCE_ID = "fill-source-id";
+    private static final String LINE_SOURCE_ID = "line-source-id";
+    private static final String CIRCLE_LAYER_ID = "circle-layer-id";
+    private static final String FILL_LAYER_ID = "fill-layer-polygon-id";
+    private static final String LINE_LAYER_ID = "line-layer-id";
+    private static final int FILL_COLOR = 0x88736357;
+    private static final int LINE_COLOR = 0xEE736357;
+    private static final int POINT_COLOR = 0xFF736357;
+    private static final int POINT_LINE_COLOR = 0xFF5B5048;
+
+    private MapView mapView;
+    private MapboxMap mapboxMap;
+
+    private Style.OnStyleLoaded callback = this::updateSources;
+    private FeatureCollection features;
+    private LatLngBounds latLngBounds;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_view_geo_shape);
+        setupToolBar();
+        setUpFeatures();
+        setUMapView(savedInstanceState);
+    }
+
+    private void setUMapView(Bundle savedInstanceState) {
```

Comment:
Should it be that? 👇 

```suggestion
    private void setUpMapView(Bundle savedInstanceState) {
```